packages/tools/client-debugger/client-debugger/src/FluidDebuggerLogger.ts

```diff
@@ -21,6 +30,57 @@ import { debuggerMessageSource, postMessagesToWindow, TelemetryEventMessage } fr
  * @internal
  */
 export class FluidDebuggerLogger extends TelemetryLogger {
+	/**
+	 * Accumulated data for Telemetry logs.
+	 */
+	private readonly _telemetryLog: ITelemetryBaseEvent[];
+
+	/**
+	 * Create an instance of this logger
+	 */
+	private static _instance: FluidDebuggerLogger;
+
+	/**
+	 * Handlers for inbound messages related to the debugger.
+	 */
+	private readonly inboundMessageHandlers: InboundHandlers = {
+		["GET_TELEMETRY_HISTORY"]: (untypedMessage) => {
+			this.postLogHistory();
+			return true;
+		},
+	};
+
+	/**
+	 * Event handler for messages coming from the window (globalThis).
+	 */
+	private readonly windowMessageHandler = (
+		event: MessageEvent<Partial<IDebuggerMessage>>,
+	): void => {
+		handleIncomingWindowMessage(event, this.inboundMessageHandlers, this.messageLoggingOptions);
+	};
+
+	/**
+	 * Posts a {@link IDebuggerMessage} to the window (globalThis).
+	 */
+	private readonly postLogHistory = (): void => {
+		postMessagesToWindow<AllTelemetryEventsMessage>(this.messageLoggingOptions, {
+			source: debuggerMessageSource,
+			type: "ALL_TELEMETRY_EVENTS",
+			data: {
+				contents: this._telemetryLog,
+			},
+		});
+	};
+
+	// #endregion
+
+	/**
+	 * Message logging options used by the debugger.
+	 */
+	private get messageLoggingOptions(): MessageLoggingOptions {
+		return { context: `FluidClientDebuggerLogger` };
+	}
```

Comment:
These options are a fixed object so we can make it a `private readonly` variable instead of a getter. That will save us a new memory allocation for the new object each time the function gets called.